packages/makeup-navigation-emitter/src/index.js

```diff
@@ -81,89 +190,141 @@ function onFocusExit() {
     }
 }
 
-function onMutation() {
-    // clear data-makeup-index on ALL items
-    clearData(this.items);
-
-    // set data-makeup-index only on filtered items (e.g. non-hidden ones)
-    setData(this.filteredItems);
-
-    if (this.index >= this.items.length) {
-        // do not use index setter, it will trigger change event
-        this._index = this.options.autoReset || this.options.autoInit;
+function onMutation(e) {
+    const fromIndex = this.index;
+    let toIndex = this.index;
+    // https://developer.mozilla.org/en-US/docs/Web/API/MutationRecord
+    const { addedNodes, attributeName, removedNodes, target, type } = e[0];
+
+    if (type === 'attributes') {
+        if (target === this.currentItem) {
+            if (attributeName === 'aria-disabled') {
```

Comment:
It's rare that conditional will ever be hit, because a disabled item can never have focus. The only time it can is if it is disabled then immediately re-enabled without it ever losing focus.

Yes, I didn't add disabled. We don't have that use case right now, but hopefully the API makes it easy enough to add it in when we need to/want to.